src/docs/arch-eviction.dox

```diff
@@ -1,15 +1,80 @@
-in-memory format to on-disk format), and then written. Clean memory
-can be freed if needed.
+@section eviction_overall Eviction
 
-Eviction cannot be triggered directly by API calls, but happens as a result
-of enough data being dirtied.
+The goal of eviction is to manage the @ref arch-cache and keep the cache in the margin set by the
+user. There is an upper limit for the total content and dirty content, whenever the cache
+goes above the target then eviction gets triggered to bring it down under the limits.
 
 Eviction is managed using \c WT_EVICT_QUEUE structures,
 each of which contains a list of \c WT_EVICT_ENTRY structures.
+
+Eviction is composed of one eviction server, 0 or more eviction worker threads and three
+shared eviction_queues (two ordinary queues and one urgent queue). The eviction server's goal
+is to find the pages that can be evicted and to do so it walks the trees, moves from one
+file to the other, and try to find evictable candidates. It makes fair judgments between
+the trees, remembers where it was in a tree, and proceeds to the next tree from there. It
+extracts all the pages that can be evicted, sorts them, and takes 1/3rd of the evictable
+candidates to do an approximate LRU and pushes them into the eviction_queues. The whole
+goal is to remove the content which is older to be evicted and put it in the queue. A user can
+configure a minimum and a maximum number of eviction worker threads to be created through
+the connection configuration string. The goal of eviction worker threads is to pop the
+pages from the queues and evict them. Eviction worker threads wait for queues to be populated,
+working with the Server through a CV signaling mechanism. They then get to work
+independently with the server, till the queues contain enough work for the candidates. There is
+a mechanism where the number of worker threads can be scaled dynamically between the range
+depending upon if we are getting more performance with more workers or getting still good
+performance and do context switching.
+
+It is possible to have just the eviction server and no worker threads. In such a case,
+the eviction server walks the tree to find pages to evict and evict these pages from the
+cache. Pages that are being marked for forced eviction go onto the urgent
+queue, and these pages take precedence over the pages from the ordinary eviction queues.
+
+If there are other threads reading the content, the page become un-evictable, and the
+Eviction server/worker threads check for that. Eviction has to lock and get exclusive
+access to the page so that after having checked if the page is evictable and starting to
+evict, a parallel reader doesn't start to access it.
+
+@section clean_dirty_data Clean vs dirty data
+- Clean refers to data in the cache that is identical to the version stored on disk.
+- Dirty refers to data in the cache that has been modified and needs to be reconciled
+  with the data stored in disk.
```

Comment:
"stored in disk." --> "stored on the disk."

Reply:
Fixed